catalog/app/containers/Bucket/PackageDialog/PackageDialog.js

```diff
@@ -154,23 +190,52 @@ export const getMetaValue = (value) =>
       )
     : undefined
 
-export function Field({ input, meta, errors, label, ...rest }) {
-  const error = meta.submitFailed && meta.error
-  const validating = meta.submitFailed && meta.validating
+export function Field({ error, helperText, validating, warning, ...rest }) {
   const props = {
+    InputLabelProps: { shrink: true },
+    InputProps: {
+      endAdornment: validating && <M.CircularProgress size={20} />,
+    },
     error: !!error,
-    label: (
-      <>
-        {error ? errors[error] || error : label}
-        {validating && <M.CircularProgress size={13} style={{ marginLeft: 8 }} />}
-      </>
-    ),
+    helperText: error || helperText,
+    ...rest,
+  }
+  return <M.TextField {...props} />
+}
+
+export function PackageNameInput({ errors, input, meta, validating, ...rest }) {
```

Comment:
why we need to special-case these inputs?

Reply:
They have different logic now: name shows error as soon as user type something in (or even on start for Update and Copy), commit message shows error after submitFailed only.

Reply:
Also NameInput has workaround for fixing infinite spinner, CommitMessageInput doesn't have it

Reply:
ok makes sense